mono/utils/os-event-unix.c

```diff
@@ -0,0 +1,353 @@
+/*
```

Comment:
This is the exact same code as `w32event-unix.c` and `w32handle.c` code combined to extract our behaviour of `System.Threading.EventWaitHandle`